getting_started.md

```diff
@@ -52,74 +52,84 @@ using Flux
 ```
 <br>
 
-### Step 2: Create some train data
-For this example, create some random train data `x` and `y` arrays:
+### Step 2: Create the training data
+First, we'll write a function that generates our "true" data. We'll use to use Flux to recover `W_actual` and `b_actual` by looking only at examples of the `actual_model` function.
 
 ```julia
-x = rand(5)
-y = rand(2) 
+W_actual = [1 2 3 4 5;
+            5 4 3 2 1]
+b_actual = [- 1.0; -2.0]
+actual_model(x) = W_actual * x + b_actual
 ```
 <br>
 
+Next, we generate our training data by passing random vectors into the true model. We'll also add Gaussian noise using `randn()` so that it's not *too* easy for Flux to figure out the model.
+
+```julia
+x_train = [ 5 .* rand(5) for _ in 1:10_000 ]
+y_train = [ actual_model(x) + 0.2 .* randn(2) for x in x_train 
+```
+
+In most real machine learning scenarios, you would not have access to `actual_model`, and would just get `x_train` and `y_train` as input data.
+
 ### Step 3: Define your model
 
-Define a simple regression model by defining the following function:
+Next, we define the model we want to use to learn the data. We'll use the same form that we used for our training data:
 
 ```julia
-model(x) = W*x .+ b
+training_model(x) = W_learned*x + b_learned
```

Comment:
```suggestion
training_model(x) = W_learned*x .+ b_learned
```
`.+` is good broadcasting hygiene (`+` on arrays calls it behind the scenes). I'd also lean towards calling `training_model` just `model` and renaming `actual_model` to better represent its role as a ground truth function (side note: is "true model" an oxymoron?)

Reply:
I was suspicious of this since broadcasting is partially what allowed #101 to slip through in the first place (without broadcasting, the dimensions would have failed to match), but you're right in that it's probably better to set a good example here than it is to guard against a one-off case.

Perhaps `model` and `ground_truth`?

Reply:
Perhaps there is no true model, only a most-useful model.